test/database/projectComponent.js

```diff
@@ -0,0 +1,249 @@
+      return expect(ProjectComponent.update(userTwo.id, component.id, {description: 'foo'})).to.be.rejectedWith('Cannot edit a component in a project you are not a part of');
+    });
+    it('Should reject when updating a component that does not exist', () => {
+      return expect(ProjectComponent.update(userOne.id, 123456, {description: 'foo'})).to.be.rejectedWith('Component does not exist');
+    });
+  });
+
+  describe('delete()', () => {
+    let component;
+    beforeEach(() => {
+      return ProjectComponent.create({
+        userId: userOne.id,
+        projectId: project.id,
+        name: 'test component',
+        type: 'image',
+        resourceUrl: 'google.com'
+      })
+        .then((projectComponent) => {
+          component = projectComponent;
+        });
+    });
+
+    it('Should delete as project owner when all parameters are valid', () => {
+      return ProjectComponent.delete(userOne.id, component.id)
+        .then((response) => {
+          expect(response).to.equal(true);
+        });
+    });
+    it('Should delete as project contributor when all parameters are valid', () => {
+      return Project.addContributor({ownerId: userOne.id, contributorId: userTwo.id, projectId: project.id})
+        .then(() => {
+          return ProjectComponent.delete(userTwo.id, component.id)
+            .then((response) => {
+              expect(response).to.equal(true);
+            });
+        });
+    });
+    it('Should reject when deleting and not an owner or contributor', () => {
+      return expect(ProjectComponent.delete(userTwo.id, component.id)).to.be.rejectedWith('Cannot delete a component in a project you are not a part of');
+    });
+    it('Should reject when deleting a component that does not exist', () => {
+      return expect(ProjectComponent.delete(userOne.id, 123456)).to.be.rejectedWith('Component does not exist');
+    });
+  });
+
+  describe('getByProject()', () => {
+    let component;
+    beforeEach(() => {
+      return ProjectComponent.create({
+        userId: userOne.id,
+        projectId: project.id,
+        name: 'test component',
+        type: 'image',
+        resourceUrl: 'google.com'
+      })
+        .then((projectComponent) => {
+          component = projectComponent;
+        });
+    });
+
+    it('Should resolve to an array of components when projectId is valid', () => {
+      return ProjectComponent.getByProject(project.id)
+        .then((components) => {
+          expect(components).to.be.a('array');
+          expect(components.length).to.equal(1);
+          expect(components[0].name).to.equal('test component');
+          return ProjectComponent.getByProject(project.id);
+        });
+    });
+    it('Should reject if userId does not map to an existing project', () => {
+      return expect(ProjectComponent.getByProject(1234)).to.be.rejectedWith('Project does not exist');
+    });
+  });
+
+  // TODO - Use these tests
```

Comment:
Wait why do you have this commented out? If this is not implemented yet you can simple create the skeleton. 

xdescribe('getByUser()', () => {
})